### PR TITLE
Create isPowerOfTwo.md

### LINKED ISSUE #1609: [Feature]: Add isPowerOfTwo
### Feature Name

Add -- isPowerOfTwo problem solution code in java

### Feature Description

I think this code should be added on java section folder of this repo.

### Motivation

It'll enhance learning.

### Implementation Suggestions (Optional)

_No response_

### Feature Type

Other

### Does this feature require additional resources?

- [ ] Additional Documentation
- [ ] Third-Party Libraries
- [ ] UI/UX Design
- [X] Other

### References (Optional)

_No response_

### PR DESCRIPTION
**Pull request by kjl98**
### Description
To solve this problem, we can use bit manipulation, which is very efficient. The key observation here is that a number which is a power of two has exactly one bit set to 1 in its binary representation.

Fixes #1609 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: java/isPowerOfTwo.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+---
+id: isPowerOfTwo
```
Comment (ajay-dhangar) [inline java/isPowerOfTwo.md]: replace `id: isPowerOfTwo` with `id: is-power-of-two`

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your id in good format for our `url`

**Review by ajay-dhangar — APPROVED** [review]
Done

**Comment by kjl98** [comment]
> Done

Thank you @ajay-dhangar for merging my changes.

**Pull request merged**